src/components/markdown/ProblemsList/ProblemsListItem.tsx

```diff
@@ -173,6 +174,34 @@ export default function ProblemsListItem(
   const [copied, setCopied] = React.useState(false);
   const tippyRef = useRef<Instance>();
 
+  const { solutions, currentUserSolutions } = useUserSolutionsForProblem(
+    problem
+  );
```

Comment:
calling `useUserSolutionsForProblem` downloads all the submissions for every problem on a given page. This adds ~100kb of network bandwidth for Intro to Prefix Sums page.

Ideally the number of user solutions for a particular problem would be stored in a separate document (with just the number of user solutions) and this document would be updated every time a user submission is added or deleted. But this seems like it would require a lot of time to implement so I'm fine with the current implementation :D